xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/user/impl/xwiki/MyBasicAuthenticator.java

```diff
@@ -73,6 +95,8 @@ public boolean processLogin(String username, String password, String rememberme,
             // make sure the Principal contains wiki name information
             if (!StringUtils.contains(principal.getName(), ':')) {
                 principal = new SimplePrincipal(context.getWikiId() + ":" + principal.getName());
+
+                getObservationManager().notify(new UserAuthenticatedEvent(getContextUserReference(principal)), null);
```

Comment:
The point of this if is to modify the principal, but in both cases the user is authenticated.